app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt

```diff
@@ -403,11 +417,34 @@ open class UserViewModel(
             updateUser(
                 updatedSender,
                 onSuccess = {
-                  // Optimistically update the state flow
-                  _friendRequests.value =
-                      _friendRequests.value.filterNot { it.id == friendRequest.id }
-                  // Clear the friend request after successful updates
-                  friendRequestRepository.deleteRequest(friendRequest.id, {}, {})
+                  // Mark the request as accepted to trigger the sender's notification
+                  val acceptedRequest = friendRequest.copy(accepted = true)
+                  friendRequestRepository.createRequest(
+                      acceptedRequest,
+                      onSuccess = {
+                        // Remove the old request from this device's local state immediately
+                        _friendRequests.value =
+                            _friendRequests.value.filterNot { it.id == friendRequest.id }
+
+                        // After a short delay, delete the accepted request from Firestore
+                        // This gives the sender time to see the acceptance state and show a
+                        // notification
+                        viewModelScope.launch {
+                          kotlinx.coroutines.delay(1000)
+                          friendRequestRepository.deleteRequest(
```

Comment:
Does this work if the other user is offline(without connection) / doesn't have the app open?

Reply:
For now, since when we close the app we are no longer logged in you do not receive the notification that somebody accepted your friend request. But the 'kotlinx.coroutines.delay(1000)' makes sure that the friend request logic is updated correctly and that the notification is received if the app is opened. To implement notifications for when the app is not open I would need to use a push notification service like Firebase Cloud Messaging, which can be done, but I believe it should be the topic of another PR.

Reply:
Hmm, maybe you are right, but I was running under the impression that when we close the app the `Firebase.auth.uid` is still preserved so we don't necessarily log in (this is why the offline mode still works when you restart the app). 

Another thing you might have missed / might need to introduce in the next PR is that if you click on the notification for a friend request, the app should open directly on your profile so you can accept the friend request.

Reply:
While you are right about the fact that Firebase.auth.uid is preserved, for the os to be able to handle background notifications, we need to implement a service like Firebase Cloud Messaging. Regarding your second point, that's for the suggestion, I will make sure to implement it as soon as Firebase Cloud Messaging is implemented.